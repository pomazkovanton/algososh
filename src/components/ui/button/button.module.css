.button {
  display: flex;
  align-items: center;
  justify-content: center;

  min-width: 130px;
  max-height: 60px;
  padding: 18px 24px 19px;

  background-color: var(--bg-button-default);
  border: none;
  border-radius: var(--common-border-radius);
  outline: none;

  transition: var(--common-transition);
}

.button:hover:enabled {
  cursor: pointer;
  background-color: var(--bg-button-hover);
}

.button:active:enabled {
  cursor: pointer;
  color: var(--text-color-secondary);
  background-color: var(--bg-button-active);
  fill: var(--text-color-secondary);
}

.button:disabled {
  background-color: var(--bg-button-disabled);
}

.big {
  width: 362px;
}

.small {
  width: 175px;
}

.loader:disabled {
  background-color: var(--bg-button-default);
}

.loader_icon {
  animation: rotation 1s linear infinite normal forwards running;
}

@keyframes rotation {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
