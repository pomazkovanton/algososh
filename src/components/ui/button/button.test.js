import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from "./button";

describe('Компонент <Button />', () => {
    it('отрисовывается с текстом', () => {
        const tree = renderer
          .create(<Button  text="Кнопка"/>)
          .toJSON();
          expect(tree).toMatchSnapshot();
    }); 

    it('отрисовывается без текста', () => {
        const tree = renderer
          .create(<Button />)
          .toJSON();
          expect(tree).toMatchSnapshot();
    }); 

    it('отрисовывается в сотоянии disabled', () => {
        const tree = renderer
          .create(<Button disabled/>)
          .toJSON();
          expect(tree).toMatchSnapshot();
    }); 

    it('отрисовывается с индикацией загрузки', () => {
        const tree = renderer
          .create(<Button isLoader/>)
          .toJSON();
          expect(tree).toMatchSnapshot();
    }); 
    
    it('отрабатывается вызов колбека при клике', () => {
        window.alert = jest.fn();
        render(<Button text="Кнопка" onClick={alert("Кнопка нажата!")}/>)
        const btn = screen.getByText("Кнопка");
        fireEvent.click(btn);
        expect(window.alert).toHaveBeenCalledWith('Кнопка нажата!');
    }); 
});