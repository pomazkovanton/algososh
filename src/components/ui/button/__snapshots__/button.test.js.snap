// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <Button /> отрисовывается без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент <Button /> отрисовывается в сотоянии disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент <Button /> отрисовывается с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Компонент <Button /> отрисовывается с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;
